src/components/Search/index.tsx

```diff
@@ -158,7 +165,7 @@ const indices = [
 ]
 
 // Validate against basic requirements of an ETH address
-const isValidAddress = (address) => {
+const isValidAddress = (address): boolean => {
   return /^(0x)?[0-9a-f]{40}$/i.test(address)
 }
 
```

Comment:
`connectStateResults` accepts a component as an argument. Currently, this argument is defined in the functional call. I shall prefer that I separate it out and then, pass the component as an Argument

Reply:
That way I can define proper React.FC type for the same

Reply:
Yes please! lets do that. Use `import type { StateResultsProvided } from "react-instantsearch-core"` as it props 👍🏼